File: barcode/scratch/check_braces.dart

import 'dart:io';

void main() {
  Directory('lib').listSync(recursive: true).forEach((entity) {
    if (entity is File && entity.path.endsWith('.dart')) {
      final content = entity.readAsStringSync();
      final openBraces = '{'.allMatches(content).length;
      final closeBraces = '}'.allMatches(content).length;
      if (openBraces != closeBraces) {
        print('Brace mismatch in ${entity.path}: {: $openBraces, }: $closeBraces');
      }
      final openParens = '('.allMatches(content).length;
      final closeParens = ')'.allMatches(content).length;
      if (openParens != closeParens) {
        print('Paren mismatch in ${entity.path}: (: $openParens, ): $closeParens');
      }
    }
  });
}

File: barcode/scratch/check_braces.py

import os

def check_braces(directory):
    for root, dirs, files in os.walk(directory):
        for file in files:
            if file.endswith('.dart'):
                path = os.path.join(root, file)
                with open(path, 'r', encoding='utf-8') as f:
                    content = f.read()
                    open_braces = content.count('{')
                    close_braces = content.count('}')
                    if open_braces != close_braces:
                        print(f"Brace mismatch in {path}: {{: {open_braces}, }}: {close_braces}")
                    
                    open_parens = content.count('(')
                    close_parens = content.count(')')
                    if open_parens != close_parens:
                        print(f"Paren mismatch in {path}: (: {open_parens}, ): {close_parens}")

if __name__ == "__main__":
    check_braces('lib')
